### PR TITLE
Fix memory issues in tests

### PR DESCRIPTION
**Pull request by DN6**
# What does this PR do?
A lot of the memory issues in tests can be solved by 

1. Enabling model cpu offload
2. Using a different (smaller) model

This PR makes the necessary changes for these tests to pass. Additionally, it fixes a small bug in `enable_model_cpu_offload` when `self._offload_gpu_id` is not properly set. 

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/pipelines/pipeline_utils.py [hunk 1]
```diff
@@ -498,6 +498,7 @@ class DiffusionPipeline(ConfigMixin, PushToHubMixin):
     _exclude_from_cpu_offload = []
     _load_connected_pipes = False
     _is_onnx = False
+    _offload_gpu_id = None
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/pipeline_utils.py:501]: is this necessary? `_offload_gpu_id` should not be a class variable IMO

Comment (DN6) [reply src/diffusers/pipelines/pipeline_utils.py:501]: This line would throw an error if it isn't defined anywhere. 
https://github.com/huggingface/diffusers/blob/514d2501c3f2d361a5fbcccbb60e5e88b381efa9/src/diffusers/pipelines/pipeline_utils.py#L1364

I added it as a class variable based on this comment

```
# _offload_gpu_id should be set to passed gpu_id (or id in passed `device`) or default to previously set id or default to 0
```
Assumed that `previously set id`  meant that the user could set it before hand.

Comment (patrickvonplaten) [reply src/diffusers/pipelines/pipeline_utils.py:501]: Ah yeah you're 100% right. Could we instead though do the following:

```py
self._offload_gpu_id = gpu_id or torch_device.index or getattr(self, "_offload_gpu_id", 0)
```

both here https://github.com/huggingface/diffusers/blob/514d2501c3f2d361a5fbcccbb60e5e88b381efa9/src/diffusers/pipelines/pipeline_utils.py#L1364 and here: https://github.com/huggingface/diffusers/blob/514d2501c3f2d361a5fbcccbb60e5e88b381efa9/src/diffusers/pipelines/pipeline_utils.py#L1449

**Comment by DN6** [comment]
@patrickvonplaten Ready for another review

**Pull request merged**